sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/ResultReturner.java

```diff
@@ -155,197 +158,161 @@ protected Type elementType(StatementContext ctx) {
-            return ctx.findElementTypeFor(returnType).orElse(returnType);
+        protected Type elementType(StatementContext ctx) {
+            return elementType;
         }
     }
 
-    static class SingleValueReturner extends ResultReturner
-    {
-        private final Type returnType;
+    static class ConsumerResultReturner extends ResultReturner {
+        private final int consumerIndex;
+        private final Type elementType;
 
-        SingleValueReturner(Type returnType)
-        {
-            this.returnType = returnType;
+        ConsumerResultReturner(Method method, int consumerIndex) {
+            this.consumerIndex = consumerIndex;
+            elementType = method.getGenericParameterTypes()[consumerIndex];
         }
 
         @Override
-        protected Object mappedResult(ResultIterable<?> iterable, StatementContext ctx)
-        {
-            return checkResult(iterable.findFirst().orElse(null), returnType);
+        protected Void mappedResult(ResultIterable<?> iterable, StatementContext ctx) {
+            @SuppressWarnings("unchecked")
+            Consumer<Object> consumer = (Consumer<Object>)
+                ctx.getConfig(SqlObjectStatementConfiguration.class).getArgs()[consumerIndex];
+            iterable.forEach(consumer);
+            return null;
         }
 
         @Override
-        protected Object reducedResult(Stream<?> stream, StatementContext ctx) {
-            return checkResult(stream.findFirst().orElse(null), returnType);
+        protected Void reducedResult(Stream<?> stream, StatementContext ctx) {
+            @SuppressWarnings("unchecked")
+            Consumer<Object> consumer = (Consumer<Object>)
+                ctx.getConfig(SqlObjectStatementConfiguration.class).getArgs()[consumerIndex];
+            stream.forEach(consumer);
+            return null;
         }
 
         @Override
-        protected Type elementType(StatementContext ctx)
-        {
-            return returnType;
-        }
-    }
-
-    private static Object checkResult(Object result, Type type) {
-        if (result == null && getErasedType(type).isPrimitive()) {
-            throw new IllegalStateException("SQL method returns primitive " + type + ", but statement returned no results");
+        protected Type elementType(StatementContext ctx) {
+            return elementType;
         }
-        return result;
     }
 
-    static class ResultIterableReturner extends ResultReturner
-    {
-
-        private final Type elementType;
-
-        ResultIterableReturner(Type returnType)
-        {
-            // extract T from Query<T>
-            elementType = GenericTypes.findGenericParameter(returnType, ResultIterable.class)
-                    .orElseThrow(() -> new IllegalStateException(
-                            "Cannot reflect ResultIterable<T> element type T in method return type " + returnType));
-        }
-
+    static class VoidReturner extends ResultReturner {
```

Comment:
Why reorder the inner classes? Makes it difficult to review

Reply:
I've widely seen it considered good practice to define inner classes/interfaces/enums at the bottom of the parent class to keep them out of the way of the class logic

Reply:
I get that, but why change the order of the inner classes themselves?

Reply:
I may not really have paid attention to that, having to sift through hundreds of violations ;_;